lgc/interface/lgc/Builder.h

```diff
@@ -204,8 +204,7 @@ class Builder : public BuilderBase {
 
   // Create code to calculate the dot product of two integer vectors, with optional accumulator, using hardware support
   // where available.
-  // The accumulator input must be i32; use a value of 0 for no accumulation.
-  // The result type is i32.
+  // Use a value of 0 for no accumulation.
```

Comment:
Please document that the result type is the same as the accumulator type, and what restrictions there are on the result and accumulator type relative to the input element type. Can I have <4 x i64> inputs and i8 result?

Please document that the result saturates.